tests/unit/functions.tcl

```diff
@@ -1132,4 +1164,38 @@ start_server {tags {"scripting"}} {
         r function flush
         r function stats
     } {running_script {} engines {LUA {libraries_count 0 functions_count 0}}}
+
+    test {FUNCTION - function test empty engine} {
+         catch {r function load replace {#! name=test
+            redis.register_function('foo', function() return 1 end)
+        }} e
+        set _ $e
+    } {ERR Engine not found}
+
+    test {FUNCTION - function test unknown metadata value} {
+         catch {r function load replace {#!lua name=test foo=bar
+            redis.register_function('foo', function() return 1 end)
+        }} e
+        set _ $e
+    } {ERR Invalid metadata value given: foo=bar}
+
+    test {FUNCTION - function test no name} {
+         catch {r function load replace {#!lua
+            redis.register_function('foo', function() return 1 end)
+        }} e
+        set _ $e
+    } {ERR Library name was not given}
+
+    test {FUNCTION - function test multiple names} {
+         catch {r function load replace {#!lua name=foo name=bar
+            redis.register_function('foo', function() return 1 end)
+        }} e
+        set _ $e
+    } {ERR Invalid metadata value, name argument was given multiple times}
+
+    test {FUNCTION - function test name with commas} {
+        r function load replace {#!lua name="foo"
```

Comment:
you mean "quotes" not "commas", but i'm not sure we should forbid it.
maybe we should actually support it, same as we'll later support quoted description with spaces and.

Reply:
It is supported, this test checks that it is working.